src/Xamarin.Android.Build.Tasks/Tasks/GetAndroidPackageName.cs

```diff
@@ -47,29 +45,28 @@ public class GetAndroidPackageName : AndroidTask
 
 		public override bool RunTask ()
 		{
-			// If we don't have a manifest, default to using the assembly name
-			// If the assembly doesn't have a period in it, duplicate it so it does
-			PackageName = AndroidAppManifest.CanonicalizePackageName (AssemblyName);
-
-			if (string.IsNullOrEmpty (ManifestFile) || !File.Exists (ManifestFile)) {
-				Log.LogMessage ("  PackageName: {0}", PackageName);
-				return true;
+			if (!string.IsNullOrEmpty (PackageName)) {
+				PackageName = AndroidAppManifest.CanonicalizePackageName (PackageName);
+			} else if (!string.IsNullOrEmpty (ManifestFile) && File.Exists (ManifestFile)) {
+				using var stream = File.OpenRead (ManifestFile);
+				using var reader = XmlReader.Create (stream);
+				if (reader.MoveToContent () == XmlNodeType.Element) {
+					var package = reader.GetAttribute ("package");
+					if (!string.IsNullOrEmpty (package)) {
+						PackageName = AndroidAppManifest.CanonicalizePackageName (package);
+					}
+				}
```

Comment:
what happens if PackageName is empty and the ManifestFile does not exist? Is that even possible?
We no longer fall back to the AssemblyName?

Reply:
A few lines lower if `PackageName` is still empty, it does that check last.

I was trying to make `AndroidAppManifest.CanonicalizePackageName` get called less, because it seems to do a lot, including a `Regex`:

https://github.com/xamarin/xamarin-android-tools/blob/26d65d95943b18bb9575d75c8c60ce36180a0db9/src/Xamarin.Android.Tools.AndroidSdk/AndroidAppManifest.cs#L54-L59